dwave/preprocessing/include/dwave/presolve.h

```diff
@@ -145,15 +56,134 @@ class Presolver {
     /// Return a const reference to the held constrained quadratic model.
     const model_type& model() const;
 
-    /// Return a const reference to the held postsolver.
-    const Postsolver<bias_type, index_type, assignment_type>& postsolver() const;
+    /// Return a sample of the original CQM from a sample of the reduced CQM.
+    template<class T>
+    std::vector<T> restore(std::vector<T> reduced) const {
+        return model_.restore(std::move(reduced));
+    }
 
  private:
     static constexpr double FEASIBILITY_TOLERANCE = 1.0e-6;
     static constexpr double INF = 1.0e30;
 
-    model_type model_;
-    Postsolver<bias_type, index_type, assignment_type> postsolver_;
+    // We want to control access to the model in order to track changes,
+    // so we create a rump version of the model.
+    class ModelView : private model_type {
```

Comment:
More a question than a comment:  so the ModelView will be the "model" we wish to presolve plus the additional transforms structure. That's why in each technique you're removing the stuff from postsolve, as the model view is actually holding the transformations. Did I understand correctly?

Reply:
Yup, exactly. That way we're not needing to keep two different objects consistent, we have one that just needs to be self-consistent.